src/app/auth/auth.component.ts

```diff
@@ -137,6 +139,8 @@ export class AuthComponent implements OnInit, OnDestroy {
           this.logger.info('Closing authentication window');
         }
       });
+
+    this.checkAppIsOutdated();
```

Comment:
Thanks for working on this, Sean!
There's a few issues I would like to discuss!

Correct me if I'm wrong, but the code here doesn't have the intended effect, due to a caveat in JavaScript:
```
    fetch(accessTokenUrl).then(res => res.json())
      .then(...)
      .catch(...)
      .finally(...);

    this.checkAppIsOutdated();
```

The code within the `then(...)` callbacks is actually run after `this.checkAppIsOutdated()`.
This means the version check is still done using an unauthenticated call, as the access token might not be stored before version check is done.

The second issue is that the current changes only apply to the web version of our app; the `onMessage` method is used by the web version.

I have a slightly different suggestion on how we could tackle this.
Perhaps we can conduct the version check within the `completeLoginProcess` method?
https://github.com/CATcher-org/CATcher/blob/146766cf46e24ff8289bc983def4ee50c1f077b4/src/app/auth/auth.component.ts#L179-L192

We could tweak `AuthComponent.checkAppIsOutdated` so that it returns an `Observable<boolean>`.
The observable should raise an error if the app is outdated, or if the version check could not be done.

We can then add `this.checkAppIsOutdated` to the "pipeline" in `completeLoginProcess` (along with the existing `this.phaseService.sessionSetup` , `this.githubEventService.setLatestChangeEvent()`).

This would address our issue for both electron / web version at once.

There is another benefit from this method:
It allows us to display the version related errors via the notification popup (like the popup below):
![image](https://user-images.githubusercontent.com/35621759/125183633-c249b600-e24a-11eb-834e-7c5e90869946.png)

We could remove the additional HTML in our AuthComponent that displays errors pertaining to version checks.
i.e. we can remove these views:

![image](https://user-images.githubusercontent.com/35621759/125183365-9fb69d80-e248-11eb-93a8-d5f5f9a06686.png)

![image](https://user-images.githubusercontent.com/35621759/125183526-c2958180-e249-11eb-90d6-73ee1b9b1509.png)

What do you think? I think it will be more streamlined to display all errors via the notification popup. Our AuthComponent code will also be simplified if we remove the HTML used for version related errors.

Reply:
Another option would be to remove the version check entirely from the web version, and just ensure that it is done after authentication in the desktop app. 

We can leave the error-related HTML views as they are, and they will only be used in our desktop app.
This method seems efficient to me.
Perhaps this is the best way, since the desktop app is going to be deprecated eventually?

Reply:
@anubh-v sorry for the late reply! >< 

Thank you for the comments! Just realized due to the caveat as you mentioned, in the end the unauthenticated calls would still be made 😢 

As a result, I would be agreeable to adding it into the pipeline as you mentioned while completing the login process, such that it would be confirmed that authenticated calls are made and then from which, the templates can be removed as a result. 

If we are agreeable to this, we can make the changes! thank you! 👍

Reply:
Sure @seanlowjk , yes I think it will be simpler if we end up entirely removing the HTML used to display version errors, as well as the `isReady` / `versionCheckingError` booleans.

Displaying the version error via the snackbar pop-up, would be good enough for users, and it reduces the amount of code to maintain

Reply:
Alright, sounds good to me!